modules/openthread/platform/radio.cpp

```diff
@@ -0,0 +1,492 @@
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->ClearSrcMatchShortEntry(aShortAddress);
+}
+
+otError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const otExtAddress *aExtAddress)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	otExtAddress addr;
+
+	for (size_t i = 0; i < sizeof(addr); i++)
+	{
+		addr.m8[i] = aExtAddress->m8[sizeof(addr) - 1 - i];
+	}
+
+	return psRadioSpinel->ClearSrcMatchExtEntry(addr);
+}
+
+void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	SuccessOrDie(psRadioSpinel->ClearSrcMatchShortEntries());
+}
+
+void otPlatRadioClearSrcMatchExtEntries(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	SuccessOrDie(psRadioSpinel->ClearSrcMatchExtEntries());
+}
+
+otError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->EnergyScan(aScanChannel, aScanDuration);
+}
+
+otError otPlatRadioGetTransmitPower(otInstance *aInstance, int8_t *aPower)
+{
+	otError error;
+
+	OT_UNUSED_VARIABLE(aInstance);
+	VerifyOrExit(aPower != NULL, error = OT_ERROR_INVALID_ARGS);
+	error = psRadioSpinel->GetTransmitPower(*aPower);
+
+exit:
+	return error;
+}
+
+otError otPlatRadioSetTransmitPower(otInstance *aInstance, int8_t aPower)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->SetTransmitPower(aPower);
+}
+
+otError otPlatRadioGetCcaEnergyDetectThreshold(otInstance *aInstance, int8_t *aThreshold)
+{
+	otError error;
+
+	OT_UNUSED_VARIABLE(aInstance);
+	VerifyOrExit(aThreshold != NULL, error = OT_ERROR_INVALID_ARGS);
+	error = psRadioSpinel->GetCcaEnergyDetectThreshold(*aThreshold);
+
+exit:
+	return error;
+}
+
+otError otPlatRadioSetCcaEnergyDetectThreshold(otInstance *aInstance, int8_t aThreshold)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->SetCcaEnergyDetectThreshold(aThreshold);
+}
+
+int8_t otPlatRadioGetReceiveSensitivity(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->GetReceiveSensitivity();
+}
+
+#if OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
+otError otPlatRadioSetCoexEnabled(otInstance *aInstance, bool aEnabled)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->SetCoexEnabled(aEnabled);
+}
+
+bool otPlatRadioIsCoexEnabled(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->IsCoexEnabled();
+}
+
+otError otPlatRadioGetCoexMetrics(otInstance *aInstance, otRadioCoexMetrics *aCoexMetrics)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+
+	otError error = OT_ERROR_NONE;
+
+	VerifyOrExit(aCoexMetrics != NULL, error = OT_ERROR_INVALID_ARGS);
+
+	error = psRadioSpinel->GetCoexMetrics(*aCoexMetrics);
+
+exit:
+	return error;
+}
+#endif
+
+#if OPENTHREAD_CONFIG_DIAG_ENABLE
+otError otPlatDiagProcess(otInstance *aInstance, int argc, char *argv[], char *aOutput, size_t aOutputMaxLen)
+{
+	// deliver the platform specific diags commands to radio only ncp.
```

Comment:
nit: Zephyr allows only C-style comments (`/* */`), same in other places.

Reply:
resolved